cli/src/registry/publish.ts

```diff
@@ -0,0 +1,227 @@
+import { execSync } from 'child_process';
+import { createCommand } from 'commander';
+import { rmSync } from 'fs';
+import { writeFile } from 'fs/promises';
+import { tmpdir } from 'os';
+import { join } from 'path';
+import { AddRegistryEntryResult, EditRegistryEntryResult } from '../generated/control_panel';
+import { cargoProjectVersion, execAsync, readFileIntoUint8Array, toBlobString } from '../utils';
+import {
+  Application,
+  applicationToRegistryEntryMap,
+  parseRegistryApplication,
+  registryEntryToApplicationMap,
+  searchRegistry,
+} from './registry.core';
+
+const artifactsRootPath = join(__dirname, '../../../artifacts');
+
+interface PackageDedependency {
+  name: string;
+  version: string;
+}
+
+interface Package {
+  name: string;
+  description: string;
+  tags: string[];
+  categories: string[];
+  metadata: Record<string, string>;
+  version: string;
+  module: Uint8Array;
+  dependencies: PackageDedependency[];
+}
+
+const getAppRegistryEntry = async (app: Application, network: string): Promise<Package> => {
+  const upgraderVersion = cargoProjectVersion(Application.Upgrader);
+  console.log(`Creating release artifacts for project: upgrader...`);
+  execSync(`BUILD_MODE='${network}' npx nx run upgrader:create-artifacts`);
```

Comment:
If the selected app is the station then these artifacts will be generated twice, once when it's called for the station, and again when it's called for the upgrader as a dependency of the station. Is that intended?

Reply:
Initially yes, but now that i added the dependencies there is no need for that anymore, nice catch. I moved it to the case condition.